File: src/lib/theme.py
"""Resolve profile.brand.theme -> colour tokens (config/themes.json). Shared by ASS builder and card/cover renderers (via _theme.css)."""
import json
from pathlib import Path
ROOT = Path(__file__).resolve().parents[2]
def load(profile):
    themes = json.loads((ROOT/"config/themes.json").read_text())
    name = profile["brand"].get("theme", "ink-classic"); t = dict(themes[name]); t["name"] = name; return t
def css(t):
    return (f":root{{--paper:{t['paper']};--paper-2:{t['paper2']};--ink:{t['ink']};--muted:{t['muted']};--line:{t['line']};"
            f"--accent:{t['accent']};--accent-soft:{t['accentSoft']};--accent-on:{t['accentOn']};--mode:{t['mode']}}}\n")
def write_css(profile):
    t = load(profile); (ROOT/"templates/cards/_theme.css").write_text(css(t)); return t

File: src/lib/theme.js
// Same as theme.py for the node steps: writes templates/cards/_theme.css and returns the token set.
import fs from "node:fs"; import path from "node:path"; import { fileURLToPath } from "node:url";
const ROOT = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "..", "..");
export function loadTheme(profile) {
  const themes = JSON.parse(fs.readFileSync(path.join(ROOT, "config/themes.json"), "utf8"));
  const name = profile.brand.theme || "ink-classic"; const t = { ...themes[name], name };
  fs.writeFileSync(path.join(ROOT, "templates/cards/_theme.css"),
    `:root{--paper:${t.paper};--paper-2:${t.paper2};--ink:${t.ink};--muted:${t.muted};--line:${t.line};--accent:${t.accent};--accent-soft:${t.accentSoft};--accent-on:${t.accentOn}}\n`);
  return t;
}
